src/tags/control/Number.js

```diff
@@ -72,80 +72,113 @@ const Model = types
+        }
+        if (isDefined(self.max) && value > self.max) {
+          errors.push(`Value must be less that or equal to ${self.max}`);
+        }
+        if (isDefined(self.step)) {
+          const step = parseFloat(self.step);
+          const basis = isDefined(self.min) ? +self.min : 0;
+          const delta = (value - basis) % step;
 
-        for (const reg of object?.allRegs ?? []) {
-          // add result with default value to every region of related object without number yet
-          if (!reg.results.some(r => r.from_name === self)) {
-            reg.results.push({
-              area: reg,
-              from_name: self,
-              to_name: object,
-              type: self.resultType,
-              value: {
-                [self.valueType]: +self.defaultvalue,
-              },
-            });
+          if (delta !== 0) {
+            errors.push(`The two nearest valid values are ${value - delta} and ${value - delta + step}`);
           }
         }
-      } else {
+        if (errors.length) {
+          InfoModal.warning(`Number "${value}" is not valid: ${errors.join(', ')}.`);
+          return false;
+        }
+        return true;
+      },
+      getSelectedString() {
+        return self.number + ' star';
+      },
+
+      needsUpdate() {
+        if (self.result) self.number = self.result.mainValue;
+        else self.number = null;
+      },
+
+      beforeSend() {
+        if (!isDefined(self.defaultvalue)) return;
+
+        // let's fix only required perRegions
+        if (self.perregion && self.required) {
+          const object = self.toNameTag;
+
+          for (const reg of object?.allRegs ?? []) {
+          // add result with default value to every region of related object without number yet
+            if (!reg.results.some(r => r.from_name === self)) {
+              reg.results.push({
+                area: reg,
+                from_name: self,
+                to_name: object,
+                type: self.resultType,
+                value: {
+                  [self.valueType]: +self.defaultvalue,
+                },
+              });
+            }
+          }
+        } else {
         // add defaultValue to results for top-level controls
-        if (!isDefined(self.number)) self.setNumber(+self.defaultvalue);
-      }
-    },
+          if (!isDefined(self.number)) self.setNumber(+self.defaultvalue);
+        }
+      },
 
-    unselectAll() {},
+      unselectAll() {},
 
-    setNumber(value) {
-      self.number = value;
-      self.updateResult();
-    },
+      setNumber(value) {
+        self.number = value;
+        self.updateResult();
+      },
 
-    onChange(e) {
-      const value = +e.target.value;
+      onChange(e) {
+        const value = +e.target.value;
 
-      if (!isNaN(value)) self.setNumber(value);
-    },
+        if (!isNaN(value)) {
+          self.setNumber(value);
+          e.target.value = isDefined(self.number) ? self.number : '';
```

Comment:
why do we need this?

Reply:
It should prevent having values with leading zeros. That was annoing.